src/vs/workbench/contrib/terminal/browser/xterm/cognisantCommandTrackerAddon.ts

```diff
@@ -151,12 +151,13 @@ export class CognisantCommandTrackerAddon extends CommandTrackerAddon {
 	}
 
 	get cwds(): string[] {
-		const cwds = [];
-		const sorted = new Map([...this._cwds.entries()].sort((a, b) => b[1] - a[1]));
-		for (const [key,] of sorted.entries()) {
-			cwds.push(key);
-		}
-		return cwds;
+		return Array.from(new Map([...this._cwds.entries()].sort((a, b) => a[1] - b[1]))).map(s => s[0]);
+	}
```

Comment:
Might be a good case to use a generator function? Currently this will create a new array (Array.from), a new map (new Map), and another array (Array.map).